### PR TITLE
automate documentation upload to website

### PR DESCRIPTION
**Pull request by sandro-elsweijer**
**_Describe your changes here:_**

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

- [x] The author added a BSD statement to `doc/` (or already has one)
- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [x] All tests pass (in various configurations, this should be executed automatically in a github action)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The reviewer executed the new code features at least once and checked the results manually
- [ ] The code is covered in an existing or new test case
- [ ] New tests use the Google Test framework
- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [ ] Testing of this template: If you feel something is missing from this list, contact the developers

### CODE REVIEW

File: .github/workflows/update_documentation.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+name: "Upload documentation to website"
```
Comment (Davknapp) [inline .github/workflows/update_documentation.yml:1]: Please add a copyright statement at the beginning (our typical file-header)

File: .github/workflows/update_documentation.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    runs-on: ubuntu-latest
+    
+    timeout-minutes: 20
+    steps:
+#
+# Setup and bootstrap
+#
+    - uses: actions/checkout@v1
+    - name: Install mpich
```
Comment (Davknapp) [inline .github/workflows/update_documentation.yml:25]: Why do we need mpi to build the documentation?

File: .github/workflows/update_documentation.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      run: mkdir build_debug && cd build_debug && ../configure $CONFIG_DEBUG CFLAGS="$CFLAGS_var" CXXFLAGS="$CXXFLAGS_var"
+    - name: OnFailUploadLog
+      if: failure()
+      uses: actions/upload-artifact@v1
+      with:
+        name: config_debug_MPI.log
+        path: build_debug/config.log
+    - name: make
+      run: cd build_debug && make -j V=0 CFLAGS="$CFLAGS_var -Werror" CXXFLAGS="$CXXFLAGS_var -Werror"
+    - name: make check
```
Comment (Davknapp) [inline .github/workflows/update_documentation.yml:64]: This workflow should not check the implementation in my opinion. But it should only run, when all tests have passed. To ensure it, we can use the needs keyword, creating a dependancy between workflows: https://docs.github.com/en/actions/using-workflows/about-workflows#creating-dependent-jobs

**Review by Davknapp — CHANGES_REQUESTED** [review]

File: .github/workflows/update_documentation.yml (hunk 3 above)
Comment (Davknapp) [reply .github/workflows/update_documentation.yml:64]: My bad, what I actually ment was: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run

Comment (sandro-elsweijer) [reply .github/workflows/update_documentation.yml]: The workflow is meant to sync the documentation to the actual state of the main branch. Therefore, it is irrelevant, if the tests pass.

Comment (Davknapp) [reply .github/workflows/update_documentation.yml]: But we shouldn't update a documentation of a crashing version.

**Review by Davknapp — CHANGES_REQUESTED** [review]

File: .github/workflows/add_release_documentation.yml [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+    # build config vars
+    - name: build CFLAGS and CXXFLAGS variables
+      if: ${{ env.MINOR_RELEASE == 'true' }}
+      run: echo CFLAGS_var="-Wall -pedantic -O3" >> $GITHUB_ENV
+        && echo CXXFLAGS_var="-Wall -pedantic -O3" >> $GITHUB_ENV
+    - name: build config variables
+      if: ${{ env.MINOR_RELEASE == 'true' }}
+      run: export CONFIG_OPTIONS="--without-blas ${LESS_TEST_OPTION}"
+           && export CONFIG_SERIAL_DEBUG="$CONFIG_OPTIONS --enable-debug --with-sc=$SC_SERIAL_DEBUG/install --with-p4est=$P4EST_SERIAL_DEBUG/install"
```
[the remark this answers is not in the document]
Comment (sandro-elsweijer) [reply .github/workflows/add_release_documentation.yml:91]: You are right, this was a copy and paste mistake

File: .github/workflows/update_documentation.yml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+           && export CONFIG_SERIAL_DEBUG="$CONFIG_OPTIONS --enable-debug --with-sc=$SC_SERIAL_DEBUG/install --with-p4est=$P4EST_SERIAL_DEBUG/install"
+           && echo CONFIG_OPTIONS="$CONFIG_OPTIONS" >> $GITHUB_ENV
+           && echo CONFIG_SERIAL_DEBUG="$CONFIG_SERIAL_DEBUG" >> $GITHUB_ENV
+    - name: Check vars
+      run: echo "[$CONFIG_SERIAL_DEBUG]"
+# configure and generate doxygen
+    - name: check debugging mode
+      run: echo "Checking debug mode"
+    - name: configure
+      run: mkdir build_debug && cd build_debug && ../configure $CONFIG_DEBUG CFLAGS="$CFLAGS_var" CXXFLAGS="$CXXFLAGS_var"
```
[the remark this answers is not in the document]
Comment (sandro-elsweijer) [reply .github/workflows/update_documentation.yml]: Also a copy paste mistake

File: .github/workflows/update_documentation.yml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    - name: bootstrap
+      run: ./bootstrap
+
+    # build config vars
+    - name: build CFLAGS and CXXFLAGS variables
+      run: echo CFLAGS_var="-Wall -pedantic -O3" >> $GITHUB_ENV
+        && echo CXXFLAGS_var="-Wall -pedantic -O3" >> $GITHUB_ENV
+    - name: build config variables
+      run: export CONFIG_OPTIONS="--without-blas ${LESS_TEST_OPTION}"
+           && export CONFIG_SERIAL_DEBUG="$CONFIG_OPTIONS --enable-debug --with-sc=$SC_SERIAL_DEBUG/install --with-p4est=$P4EST_SERIAL_DEBUG/install"
```
[the remark this answers is not in the document]
Comment (sandro-elsweijer) [reply .github/workflows/update_documentation.yml:65]: see above

**Review by Davknapp — APPROVED** [review]

**Pull request merged**